packages/eslint-plugin-next/lib/utils/url.js

```diff
@@ -42,7 +42,11 @@ function getUrlFromPagesDirectories(urlPrefix, directories) {
           (url) => `^${normalizeURL(url)}$`
         )
     )
-  ).map((urlReg) => new RegExp(urlReg))
+  ).map((urlReg) => {
+    // /foo/[bar] filees fails without this.
+    urlReg = urlReg.replace(/\[/g, "\\[");
+    new RegExp(urlReg)
```

Comment:
```suggestion
    return new RegExp(urlReg)
```
Should we escape all regex characters here like we do here? https://github.com/vercel/next.js/blob/acf65f92dc7ce459a25ac5a9bc668300e8cff28b/packages/next/shared/lib/router/utils/route-regex.ts#L9-L11

Reply:
we definitely should. I'll give it a try and update this

Reply:
I moved the logic around to handle this, we don't need to handle the escaping of all the items

Reply:
added more tests to check all permutations and combinations